### DISCUSSION TITLE
Whitelisting IP+Ports combinations (IPEndPoint)

### DISCUSSION THREAD

**Discussion by JVemon** [jchristn/WatsonTcp#203] [Ideas]
The current API allows for automatic rejection of IPs that aren't whitelisted. Perhaps we can consider extending it to also allow the whitelisting of IP+Port combinations (or well, an `IPEndPoint`)?

This may be a strange request since typically client connections pick a random port. My use case in particular is as follows:

---
An online game cluster with Authentication and Gameplay nodes, where the Gameplay nodes only allow connections from specific IP+Port combinations - such a whitelist is updated by the Authentication nodes, who notify the Gameplay servers whom to expect and from where (plus some shared access token they have to provide). So, players can use a random port when connecting to the Authentication node, but afterwards they should disconnect and re-use it when connecting to the actual game world.

---

I cannot think of many other useful scenarios, so I'm not sure if the library would truly benefit from something as oddly specific. Besides, I do have concerns about clients being able to select their desired local port on some platforms, such as game consoles.

**Comment by jchristn** [comment]
Hi @JVemon I considered implementing this but thought it would be fairly narrow in scope.  However if you want to put together a PR with amendments to server settings to include ```PermittedIPEndPoints``` and ```BlockedIPEndPoints```, along with changes in the connection acceptance loop, I'd be happy to integrate!
